### PR TITLE
Adding the Cart to search #1092

### LINKED ISSUE #1092: Cart for selection is not visible in search results page
**Description**:
The search page is not showing the cart at the top, so when selecting some items, it is only possible to see that the warning selection has been updated, but not the summary/total number of selected items as shown for other views.

It would be good to harmonise this.

Related to #1031 

Screenshot 2022-01-28 at 17 24 05

**Acceptance criteria**:
- [ ] Show selection cart in search view

### PR DESCRIPTION
**Pull request by joshuadkitenge**
## Description
Adding the cart to search.

- Adding the cart to the end of the search app bar  
- Moving aria-labels to the default.json for cart aria-labels 

## Testing instructions
Add a set up instructions describing how the reviewer should test the code
Check the cart colour in all the different themes 

- [x] Review code
- [x] Check Actions build
- [x] Review changes to test coverage
- [ ] {more steps here}

## Agile board tracking
closes #1092

### CODE REVIEW

**Comment by joshuadkitenge** [comment]
search cart

File: packages/datagateway-search/src/searchPageCardView.component.test.tsx [hunk 1]
```diff
@@ -67,6 +70,8 @@ describe('SearchPageCardView', () => {
     props = {
       onTabChange: onTabChange,
       currentTab: 'investigation',
+      cartItems: useCart,
```
Comment (louise-davies) [inline packages/datagateway-search/src/searchPageCardView.component.test.tsx:73]: technically this should be an array right? I know it probably doesn't affect the test but the type here is wrong

File: packages/datagateway-search/src/searchPageTable.component.tsx [hunk 2]
```diff
@@ -117,9 +131,32 @@ function a11yProps(index: string): React.ReactFragment {
...
+            props.cartItems.length > 0 ? props.cartItems.length : null
+          }
+          color="primary"
+        >
+          <ShoppingCartIcon />
+        </Badge>
+      </IconButton>
+    </div>
+  );
+};
```
Comment (louise-davies) [inline packages/datagateway-search/src/searchPageTable.component.tsx:156]: This code is repeated - can we extract it into it's own tiny component file to reduce code repetition?

Comment (joshuadkitenge) [reply packages/datagateway-search/src/searchPageTable.component.tsx:156]: ok, I'll moving this into datagateway-common

**Comment by joshuadkitenge** [comment]
I have moved the cart button component into common, and I have exported it into both data view and card view

**Comment by joshuadkitenge** [comment]
- @louise-davies  wait for https://github.com/ral-facilities/datagateway/pull/1109 to be merged before review this PR as the failing tests are due to #1109

**Review by louise-davies — APPROVED** [review]
Looks good and works correctly :)

**Pull request merged**